inference-engine/src/inference_engine/include/openvino/runtime/infer_request.hpp

```diff
@@ -0,0 +1,221 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header file that provides wrapper classes for infer requests and callbacks.
+ *
+ * @file infer_request.hpp
+ */
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "variable_state.hpp"
+
+namespace InferenceEngine {
+class IInferRequestInternal;
+class Blob;
+}  // namespace InferenceEngine
+
+namespace ov {
+namespace runtime {
+/**
+ * @brief This is an interface of asynchronous infer request
+ *
+ * It can throw exceptions safely for the application, where it is properly handled.
+ */
+class INFERENCE_ENGINE_API_CLASS(InferRequest) {
```

Comment:
Am I right that this PR is intersected with https://github.com/openvinotoolkit/openvino/pull/7151?

Reply:
No. should be rebased do master